newrelic/resource_newrelic_alert_muting_rule.go

```diff
@@ -26,6 +27,68 @@ func validateMutingRuleConditionAttribute(val interface{}, key string) (warns []
 	v := validation.StringInSlice([]string{"accountId", "conditionId", "policyId", "policyName", "conditionName", "conditionType", "conditionRunbookUrl", "product", "targetId", "targetName", "nrqlEventType", "tag", "nrqlQuery"}, false)
 	return v(valueString, key)
 }
+func validateNaiveDateTime(val interface{}, key string) (warns []string, errs []error) {
+	valueString := val.(string)
+
+	// test conversion to desired format:
+	_, err := time.Parse("2006-01-02T15:04:05", valueString)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("%#v of %#v must be in the format 2006-01-02T15:04:05", key, valueString))
+	}
+	return
+}
+
+func scheduleSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"end_repeat": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The datetime stamp when the MutingRule schedule should stop repeating.",
+				ConflictsWith: []string{"schedule.repeat_count"},
```

Comment:
I think for this to work as client side validation, I think this needs to use the index `0` - e.g. `[]string{"schedule.repeat_count"}`. Also I'm not sure if we need to do that because it appears the API returns a helpful error message.

e.g. 

```
newrelic_alert_muting_rule.foo: Creating...

Error: A downstream error occurred., [{"extensions":{"code":"BAD_USER_INPUT","validationErrors":[{"name":"schedule","reason":"repeat_count 1 provided in addition to end_repeat 2022-06-11T12:00:00 Please remove end_repeat or repeat_count"}]},"message":"Validation Error"}]

  on main.tf line 34, in resource "newrelic_alert_muting_rule" "foo":
  34: resource "newrelic_alert_muting_rule" "foo" {
```

Reply:
I added a 0 to both and they tested fine! Nice catch! Thanks for calling this out, adding this gotcha to our onboarding doc